### ISSUE TITLE
Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.

### ISSUE THREAD

**Issue by allendav** [woocommerce/woocommerce-gateway-stripe#1017]
**Describe the bug**

The following warning occurs when running `composer install`

> Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.

**Comment by RadoslavGeorgiev** [comment]
I investigated this, and I think we should let it be for now.

`phpunit-mock-objects` is used by PHPUnit (the sole dependency in `composer.json`). We are currently using version 5 of the framework, so I tried to see what happens if I upgrade it.

I switched to `6.0` and the result is as follows:

```
composer why phpunit/phpunit-mock-objects
phpunit/phpunit  6.0.0  requires  phpunit/phpunit-mock-objects (^4.0) 
```

So PHPUnit does keep on using the same library in versions (at least) as new as `6.0`. But an upgrade even to `6.0` is a bad idea, since it [drops support for PHP 5.*](https://phpunit.de/supported-versions.html).

Basically, the package is abandoned, but is also being used in the PHPUnit version, which we need. That's why I think we should ignore it and close the issue. @allendav do you agree?

**Comment by v18** [comment]
Going to close based on Rado's analysis.
